### ISSUE TITLE
[Bug] fsevents Cannot apply hunk #1

### ISSUE THREAD

**Issue by doums** [yarnpkg/berry#931]
- [ ] I'd be willing to implement a fix

**Describe the bug**
In my company's mono repo I tried to migrate to Yarn 2. I ran `yarn set version berry` then `yarn install`, this is its output:
```
➤ YN0000: └ Completed in 18.14s
➤ YN0000: ┌ Fetch step
➤ YN0013: │ fsevents@patch:fsevents@npm%3A2.0.7#builtin<compat/fsevents>::version=2.0.7&hash=e8cd9e can't be found in the cache and will be fetched from the disk
➤ YN0001: │ Error: fsevents@patch:fsevents@npm%3A2.0.7#builtin<compat/fsevents>::version=2.0.7&hash=e8cd9e: Cannot apply hunk #1
    at c (/home/pierre/Documents/owp/aw-front/.yarn/releases/yarn-berry.js:58:888997)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /home/pierre/Documents/owp/aw-front/.yarn/releases/yarn-berry.js:58:891128
    at async o (/home/pierre/Documents/owp/aw-front/.yarn/releases/yarn-berry.js:58:888280)
    at async Object.t.applyPatchFile (/home/pierre/Documents/owp/aw-front/.yarn/releases/yarn-berry.js:58:891101)
    at async t.PatchFetcher.patchPackage (/home/pierre/Documents/owp/aw-front/.yarn/releases/yarn-berry.js:58:888035)
    at async /home/pierre/Documents/owp/aw-front/.yarn/releases/yarn-berry.js:58:887292
    at async /home/pierre/Documents/owp/aw-front/.yarn/releases/yarn-berry.js:36:570234
    at async /home/pierre/Documents/owp/aw-front/.yarn/releases/yarn-berry.js:36:570177
➤ YN0000: └ Completed in 1.36s
➤ YN0000: Failed with errors in 19.5s
```

**To Reproduce**

Sorry but I don't share any code/package.json because it's private. But I could try to adapt an share dummy stuff.
What I can tell is we are using a mono repo architecture, using yarn workspaces feature to manage it.
The root `package.json` looks something like this:
```
{
  "name": "project x",
  "private": true,
  "scripts": {
    "clean": "./clean_node.sh",
    "build_libs": "./build_libs.sh"
  },
  "workspaces": {
    "packages": [
      "workspaces/*"
    ],
    "nohoist": [
      "**/webpack",
      "**/jest"
    ]
  },
  "dependencies": {},
  "devDependencies": {
    "eslint": "^6.6.0",
    ...
    "jest": "^24.9.0",
    "prettier": "^1.19.1"
  }
}
```
Under the directory `workspaces` are our workspaces.
We have no direct dependencies on `fsevents`.

**Screenshots**

![image](https://user-images.githubusercontent.com/6359431/74229492-c6268100-4cc2-11ea-97bc-1aca07a1073e.png)

**Environment if relevant (please complete the following information):**

 - OS: Arch Linux
 - Node version: 12.12.0
 - Yarn version: 2.0.0-rc.28

**Comment by arcanis** [comment]
Hey @doums ! Thanks for taking the time to open an issue, but I'll close it as it's a duplicate of #911 and #719 🙂

The fix is already in master, you can use it by running `yarn set version from sources`. It'll also be officially published later today.

**Comment by doums** [comment]
Ok cool, thank you !

**Comment by yorch** [comment]
Hey @arcanis 

I've just starting trying to migrate a project to yarn v2, but seeing this error using `berry`, `latest` and master, please see my comment [here](https://github.com/yarnpkg/berry/issues/719#issuecomment-650470027).

Many thanks!

**Comment by SimonBackx** [comment]
Hi! I had to use `yarn set version from sources` to fix this issue, but this issue is more than a year old? Is there a specific reason why this fix isn't published yet? 

Sadly it was a bit of a mess to upgrade to yarn berry, even when the docs said it should be easy without the PnP.

**Comment by merceyz** [comment]
This issue was fixed in 2.4.0 which was released on 30.11.2020

**Comment by SimonBackx** [comment]
> This issue was fixed in 2.0.0-rc.29 which was released on 11.02.2020

Thank you for your reply. However, I kept having the issue on the latest version of berry. If I installed from sources, the issue went away.

**Comment by merceyz** [comment]
Are you sure that was for fsevents? If it was for typescript that was fixed by https://github.com/yarnpkg/berry/pull/2889 which hasn't been released yet

**Comment by SimonBackx** [comment]
Hmmm... I also got it for TypeScript I think. I'm not sure if I got it for fsevents too. 

I've switched back to Yarn v1 because I still had some other issues with workspaces (hoisting stopped working in scripts, and I didn't liked the solution https://yarnpkg.com/getting-started/qa#how-to-share-scripts-between-workspaces), which was not described in the migration guide. Somehow these issues made me doubt the stability of Yarn v2. Maybe I'll try again later. Thank you anyway!
